Marlin/src/sd/cardreader.h

```diff
@@ -172,13 +176,16 @@ class CardReader {
   static Sd2Card& getSd2Card() { return sd2card; }
 
   #if ENABLED(AUTO_REPORT_SD_STATUS)
-    static void auto_report_sd_status();
-    static inline void set_auto_report_interval(uint8_t v) {
-      TERN_(HAS_MULTI_SERIAL, auto_report_port = multiSerial.portMask);
-      NOMORE(v, 60);
-      auto_report_sd_interval = v;
-      next_sd_report_ms = millis() + 1000UL * v;
-    }
+    //
+    // SD Auto Reporting
+    //
+    #if HAS_MULTI_SERIAL
+      static serial_index_t auto_report_port;
+    #else
+      static constexpr serial_index_t auto_report_port = 0;
+    #endif
+    class AutoReportSD : public AutoReporter<auto_report_port> { void auto_report(); };
+    static AutoReportSD auto_reporter;
   #endif
```

Comment:
I think, there is no need for a template `AutoReporter` here, just set the serial_index_t as a member of the class since it's only used for PORT_REDIRECT which is a runtime check, and you can also remove the `#if HAS_MULTI_SERIAL` condition here so it'll be simpler.